src/app/core/services/scheduling.service.ts

```diff
@@ -378,10 +378,10 @@ export class SchedulingService {
   }
 
   buildReportSchedule() {
-    let iterDate = new Date(this.refTimestamp)
+    let iterDate = this.setDateTimeToMidnight(new Date(this.enrolmentDate))
     iterDate = this.setDateTimeToMidnight(iterDate)
     const yearsMillis = DefaultScheduleYearCoverage * 60000 * 60 * 24 * 365
```

Comment:
Please use the time functions

Reply:
And also everywhere in the file. A lot of hard coded values here that are not immediately apparent.